pkg/gce-pd-csi-driver/node.go

```diff
@@ -302,3 +320,55 @@ func (ns *GCENodeServer) NodeGetVolumeStats(ctx context.Context, req *csi.NodeGe
 func (ns *GCENodeServer) NodeExpandVolume(ctx context.Context, req *csi.NodeExpandVolumeRequest) (*csi.NodeExpandVolumeResponse, error) {
 	return nil, status.Error(codes.Unimplemented, fmt.Sprintf("NodeExpandVolume is not yet implemented"))
 }
+
+func (ns *GCENodeServer) GetVolumeLimits() (int64, error) {
+	client := &http.Client{}
+	// Default value 16 for shared-CPU VM
+	volumeLimits := VolumeLimit16
+
+	// Get machine-type from metadata server
```

Comment:
Instead of directly calling the http here, I think you can extend https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/master/pkg/gce-cloud-provider/metadata/metadata.go.  It looks like the compute metadata service doesn't have a method for machine type, but you could use the general Get() call.